src/rollCall/index.ts

```diff
@@ -1,18 +1,68 @@
-import { esClient } from "elasticsearch";
+import fetch from 'node-fetch';
+import { IndexReleaseRequest, CreateResolvableIndexRequest, ResolvedIndex, RollcallClient } from './types';
 
-const ALIAS = "donors";
+export default (
+  configData: {url: string, entity?: string, type?: string, shardPrefix?: string}
+  ): RollcallClient => {
+  const rootUrl = configData.url;
+  const indexEntity = configData?.entity || "donor";
+  const indexType = configData?.type || "centric";
+  const shardPrefix = configData?.shardPrefix || "pgm";
 
-const getNewIndexName = async (programShortName: string) => {
-  const indexName = `programs-${programShortName}-${Math.random()}`;
-  await esClient.indices.create({
-    index: indexName
-  });
-  return indexName;
-};
+  const createNewResolvableIndex = async (programShortName: string, cloneFromReleasedIndex?: boolean): Promise<ResolvedIndex> => {
+    const url = `${rootUrl}/indices/create`;
 
-const release = async (index: string) => Promise.resolve();
+    const req: CreateResolvableIndexRequest = {
+      shardPrefix: shardPrefix,
+      shard: await formatProgramShortName(programShortName),
+      entity: indexEntity,
+      type: indexType,
+      cloneFromReleasedIndex: cloneFromReleasedIndex || false
+  }
 
-export default {
-  getNewIndexName: getNewIndexName,
-  release: release
-};
+    const newResolvedIndex = await fetch(url, { 
+      method: 'POST',
+      body: JSON.stringify(req), 
+      headers: {  "Content-Type": "application/json" }
+    })
+    .then(res => res.json())
+    .catch(err => { throw new Error(err) }) as ResolvedIndex;
+
+    return newResolvedIndex;
+  };
+
+  const release = async (indexName: string): Promise<boolean> =>  { 
+    const url = `${rootUrl}/aliases/release`
+
+    const req = await convertIndexNameToIndexReleaseRequest(indexName);
+
+    const acknowledged = await fetch(url, 
+      {
+        method: 'POST',
+        body: JSON.stringify(req), 
+        headers: {  "Content-Type": "application/json" }
+      }
+    ).then(res => res.json())
+    .catch(err => { throw new Error(err) })  as boolean;
+
+    return acknowledged;
+ };
+
+ const convertIndexNameToIndexReleaseRequest = async (indexName: string): Promise<IndexReleaseRequest> => {
```

Comment:
if `ResolvedIndex` already has all the info, let's just pass it right back rather than parsing `indexName` lol